app/src/main/java/org/openobservatory/ooniprobe/common/service/ServiceUtil.java

```diff
@@ -79,27 +79,32 @@ public static void callCheckInAPI(Application app) {
             return;
         }
 
-
-        AbstractSuite suite = d.generateAutoRunServiceSuite.generate(config);
+        AbstractSuite suite = d.generateAutoRunServiceSuite.generate();
         ArrayList<AbstractSuite> testSuites = new ArrayList<>();
         testSuites.add(suite);
         testSuites.add(InstantMessagingSuite.initForAutoRun());
         testSuites.add(CircumventionSuite.initForAutoRun());
         testSuites.add(PerformanceSuite.initForAutoRun());
         testSuites.add(ExperimentalSuite.initForAutoRun());
-        ServiceUtil.startRunTestService(app, testSuites, false);
+        ServiceUtil.startRunTestService(app, testSuites, false, true);
+        d.generateAutoRunServiceSuite.markAsRan();
```

Comment:
Please, rename this method. `callCheckInAPI` was misleading before because the method was doing much more than just calling the check-in API. Now, the method name has become a lie. I think we should probably proceed as follows in terms of renaming:

1. `startRunTestService` with three arguments becomes `startRunTestServiceManual`
2. `callCheckInAPI` becomes `startRunTestServiceUnattended`
3. `startRunTestService` with four arguments becomes private and is named `startRunTestServiceCommon`

What do you think?